File: src/main/java/tarmorn/io/RuleWriter.java
package src.main.java.tarmorn.io;

import java.io.FileNotFoundException;
import java.io.PrintWriter;

import src.main.java.tarmorn.structure.Rule;

public class RuleWriter {
	
	public RuleWriter() {
		
	}
	
	public void write(Iterable<Rule> rules, String filepath) throws FileNotFoundException {
		
		int i = 0;
		PrintWriter pw = new PrintWriter(filepath);
		for (Rule rule : rules) {
			pw.println(rule);
			i++;
		}
		pw.flush();
		pw.close();
		System.out.println("* wrote " + i + " rules to " + filepath);
		
	}

}

File: src/main/java/tarmorn/io/RuleWriter.kt
package tarmorn.io

import tarmorn.structure.Rule
import java.io.FileNotFoundException
import java.io.PrintWriter

class RuleWriter {
    @Throws(FileNotFoundException::class)
    fun write(rules: Iterable<Rule>, filepath: String) {
        var i = 0
        val pw = PrintWriter(filepath)
        for (rule in rules) {
            pw.println(rule)
            i++
        }
        pw.flush()
        pw.close()
        println("* wrote " + i + " rules to " + filepath)
    }
}
